.github/workflows/ci.yml

```diff
@@ -51,6 +51,13 @@ jobs:
           - 'opensuse-leap-15'
         suite:
           - 'default'
+        include:
+          - suite: 'systemd-resolved'
+            os: 'fedora-latest'
+          - suite: 'systemd-resolved'
+            os: 'ubuntu-1804'
+          - suite: 'systemd-resolved'
+            os: 'ubuntu-2004'
```

Comment:
This really should include all of the Linux platforms. I tested Amazon Linux and it seems the primary problem is the fact that the `systemd-resolved` package isn't installed. I suspect the other platforms are failing for the same reason. I recommend you add that package as being required to be installed in the resource and ensure this works on all Linux platforms.

Reply:
I guess there's a question here: should this cookbook support changing the default resolver, or just configure the one that's installed by default? I think the latter is more typically what people want, and the former might be rather complicated to implement.

Reply:
I assume if you want to manage using `systemd-resolved`, you will use the resource for that and it should set it up to use that. It's up to the users to change it's default resolver if they want to.

Reply:
I purposely haven't tried with the older distros like Amazon under the pretense that they'd sort of pre-dated the default use of it being of the centos 7 era and systemd ~210. 

To me if we wanted to install it and configure the service that'd fall to other resources with the current one purely managing the config.

Reply:
Right, but users can still prefer to use this and ensuring the resource still works on those platforms is likely a good idea unless I'm misunderstanding this.

What is the process to switch from using `/etc/resolv.conf` to `systemd-resolved` btw? Is it by simply by running `systemd-resolved` or is it more than that?

Reply:
Assuming that the distribution has a separate package for systemd-resolved and only had a regular file for /etc/resolv.conf, it is probably easy. But most distributions aren't using a regular file these days, and I'm not sure how other resolv.conf tools interact with systemd-resolved if they aren't specifically disabled. NetworkManager would likely keep modifying it, for instance, which could result in some programs getting one set of nameservers from resolv.conf and others getting a different set from systemd-resolved.

Reply:
> 
> 
> Right, but users can still prefer to use this and ensuring the resource still works on those platforms is likely a good idea unless I'm misunderstanding this.
> 
> What is the process to switch from using `/etc/resolv.conf` to `systemd-resolved` btw? Is it by simply by running `systemd-resolved` or is it more than that?

`resolved` sets up a stub listener on an address in the `localhost 127.0.0.0/8` prefix and resolv.conf is updated to point to that for backwards compatibility. If you want to go back to the legacy way then (on RH/Fedora at least) you stop and disable the systemd-resolved service and restart networkmanager to overwrite the resolv.conf with an 'old' style one (or not if you used the resource added in this PR). So I presume on Debian/Ubuntu it'd be the same apart from you most likely aren't using networkmanager.

So it's 'just' a couple of service actions which i'm not sure warrants a resource creating.

Reply:
https://fedoraproject.org/wiki/Changes/systemd-resolved has a reasonable overview of what changes

Reply:
I agree that this resource should have limited scope and not have to deal with the full steps to migrate between methods. Here's what I'd propose we do:
1. If someone is using the `resolver_systemd_resolved_config`, it should also install the `systemd-resolved` package (if it exists on said platform)
2. Update documentation to be clear that this resource does not handle converting a system from the old style method to systemd. Perhaps add a few links/steps in the README on how one might do that but be clear that it only handles the config.
3. Update the test matrix/suite to include all platforms where the above conditions should pass and work

How does that sound?

Reply:
I'd prefer this resource to panic if you try to configure a resolver that isn't installed, rather than potentially partially migrate between resolvers. Also, it's kind of annoying that containers often diverge from how an actual server would be configured, is it possible to use kitchen-vagrant for this cookbook?

Reply:
It is possible to use kitchen-vagrant in GH Actions, but it will make the tests run much longer. I don't think it's really needed for the CI on this cookbook. Nothing is stopping maintainers from running vagrant locally to ensure it works properly.

I suppose having it panic instead of installing it might be an option, Another way of doing it is making a parameter which allows the resource to install it. But I kind of expect resources to install any packages they need to work properly instead of raising an error.

Reply:
I'm somewhat torn here, I think on the whole it'd be best to leave this resource as purely configuring resolved as (personally) I wouldn't expect a *_config resource to do anything else and a couple of the last table flips I've done have been to specifically remove AIO resource behaviour.

As the implementing use have to wrap this resource anyway and with the changeover being so trivial I don't see an issue in leaving this as config-only and leaving the service actions up to the wrapper. It feels to me like expecting the core `service` resource to actually install the service package as well if we did any more, we could add resources to do the installing and service action but it feels like they're such trivial tasks in this case that I don't see any benefit in doing that.

The entry in `resolv.conf` is for backwards-compat as the 'correct' way is to flip nsswitch.conf over to [nss-resolve](https://www.freedesktop.org/software/systemd/man/nss-resolve.html) instead of nss-dns, which is why resolv.conf get symlinked to `/run/systemd/resolve/stub-resolv.conf` on distros that have gone all-in (I've only come across F33 at this point but Arch may have as well).

Reply:
We **could** possible change the `nsswitch.conf` file over though.

Reply:
Let's just leave this as-is then. If a user has a problem with how this resource works, then we can address it then.

Reply:
Sounds a plan 👍